nextplace/validator/setting_weights/weights.py

```diff
@@ -33,6 +33,31 @@ def check_timer_set_weights(self) -> None:
         bt.logging.trace("📸 Time to set weights, resetting timer and setting weights.")
         self.timer = datetime.now(timezone.utc)  # Reset the timer
         self.set_weights()  # Set weights
+    
+    def adjust_scores_based_on_recent_activity(self, scores, hotkey_to_uid):
+        """
+        Adjust scores to zero for miners with fewer than 10 predictions in the last 5 days.
+        """
+        # Get the recent date threshold
+        recent_date = (datetime.now(timezone.utc) - timedelta(days=5)).strftime(ISO8601)
```

Comment:
This needs to be imported
`from nextplace.validator.utils.contants import ISO8601`